apps/avatax/src/pages/api/webhooks/order-confirmed.ts

```diff
@@ -72,23 +73,37 @@ export default wrapWithLoggerContext(
         logger.debug("Confirming order...");
 
         if (taxProviderResult.isOk()) {
-          const confirmedOrder = await taxProviderResult.value.confirmOrder(payload.order);
-
-          logger.info("Order confirmed", { confirmedOrder });
-          const client = createInstrumentedGraphqlClient({
-            saleorApiUrl,
-            token,
-          });
-
-          const orderMetadataManager = new OrderMetadataManager(client);
-
-          await orderMetadataManager.updateOrderMetadataWithExternalId(
-            payload.order.id,
-            confirmedOrder.id,
-          );
-          logger.info("Updated order metadata with externalId");
+          try {
+            const confirmedOrder = await taxProviderResult.value.confirmOrder(payload.order);
+
+            logger.info("Order confirmed", { orderId: confirmedOrder.id });
+            const client = createInstrumentedGraphqlClient({
+              saleorApiUrl,
+              token,
+            });
+
+            const orderMetadataManager = new OrderMetadataManager(client);
+
+            await orderMetadataManager.updateOrderMetadataWithExternalId(
+              payload.order.id,
+              confirmedOrder.id,
+            );
+            logger.info("Updated order metadata with externalId");
+
+            return webhookResponse.success();
+          } catch (error) {
+            logger.debug("Error confirming order", { error });
+
+            switch (true) {
+              case error instanceof TaxBadPayloadError: {
+                // @ts-ignore - Vercel uses old version of Typescript
+                return res.status(400).send(error.message);
```

Comment:
Dont pass internal error messages to the frontend. Write here another message. We don't know what data will leak from internal errors that are passed from the stack